### PR TITLE
makes goat king loot worth something

### PR DESCRIPTION
**Pull request by LazennG**
# Document the changes in your pull request
people thought meteor gun was ok for some reason so i feel i can get away with this.
Adds new item to goat king's drops called 'goat king's scanner,' a mineral scanner that detects ores a bit more often than an advanced scanner and guarantees high potency when defusing gibtonite regardless of timing.

# Wiki Documentation
under the loot section of goat king https://wiki.yogstation.net/wiki/Megafauna#King_of_Goats

goat_scaner
goat_scaner

goat king's scanner- a mineral scanner that detects ores a bit more often than an advanced scanner and guarantees high potency when defusing gibtonite regardless of timing.

# Changelog
:cl:  @cuackles 
rscadd: Added goat king loot
imageadd: added sprite for scanner  
/:cl:
thanks to @cuackles for whipping up a better sprite for the goat scanner than me in like 5 minutes

### CODE REVIEW

File: code/game/turfs/simulated/minerals.dm [hunk 1]
```diff
@@ -815,3 +818,12 @@
 	smooth_icon = 'icons/turf/smoothrocks_hard.dmi'
 	hardness = 2
 
+/turf/closed/mineral/gibtonite/proc/defuses()
```
Comment (ToasterBiome) [inline code/game/turfs/simulated/minerals.dm]: merge this proc into defuse() and make defuse take a value force_perfect, don't need two procs that do the same thing

File: code/modules/mining/equipment/mineral_scanner.dm [hunk 2]
```diff
@@ -85,3 +85,9 @@
 /obj/effect/temp_visual/mining_overlay/Initialize()
 	. = ..()
 	animate(src, alpha = 0, time = duration, easing = EASE_IN)
+
```
Comment (ToasterBiome) [inline code/modules/mining/equipment/mineral_scanner.dm:94]: move this to right after lesser

File: yogstation/code/modules/mob/living/simple_animal/hostile/retaliate/king_of_goats.dm [hunk 3]
```diff
@@ -289,7 +289,8 @@ Difficulty: Insanely Hard
 		visible_message(span_cult("\The [src] shrieks as the seal on his power breaks and he starts to break apart!"))
 		new /obj/structure/ladder/unbreakable/goat(loc)
 		new /obj/item/toy/plush/goatplushie/angry/kinggoat(loc) //If someone dies from this after beating the king goat im going to laugh
-
+		new /obj/item/t_scanner/adv_mining_scanner/goat_scanner
```
Comment (ToasterBiome) [inline yogstation/code/modules/mob/living/simple_animal/hostile/retaliate/king_of_goats.dm]:
```suggestion
		new /obj/item/t_scanner/adv_mining_scanner/goat_scanner(loc)
```

**Review by ToasterBiome — CHANGES_REQUESTED** [review]

File: code/modules/mining/equipment/mineral_scanner.dm (hunk 2 above)
Comment (ToasterBiome) [reply code/modules/mining/equipment/mineral_scanner.dm:94]: (the definition for the scanner)

**Comment by Cartlord** [comment]
pomp adour is going to become bomb adour

**Comment by LazennG** [comment]
for clarification by 'high potency' i mean the most pwoerful gibtonite possible

**Comment by ilikesaying** [comment]
> for clarification by 'high potency' i mean the most pwoerful gibtonite possible

oh dear god the destructive potential

File: code/game/turfs/simulated/minerals.dm [hunk 4]
```diff
@@ -22,7 +22,8 @@
 	var/scan_state = "" //Holder for the image we display when we're pinged by a mining scanner
 	var/defer_change = FALSE
 	var/hardness = 1 //how hard the material is, we'll have to have more powerful stuff if we want to blast harder materials.
-
+	var/forceperfect = FALSE
```
Comment (ToasterBiome) [inline code/game/turfs/simulated/minerals.dm]:
```suggestion
```

never used

File: code/game/turfs/simulated/minerals.dm [hunk 5]
```diff
@@ -695,9 +696,12 @@
 	. = ..()
 
 /turf/closed/mineral/gibtonite/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/t_scanner/adv_mining_scanner/goat_scanner))
+		user.visible_message(span_notice("[user] holds [I] to [src]..."), span_notice("[I] locates where to cut off the chain reaction and stops it."))
+		defuse(forceperfect = TRUE)
 	if(istype(I, /obj/item/mining_scanner) || istype(I, /obj/item/t_scanner/adv_mining_scanner) && stage == 1)
 		user.visible_message(span_notice("[user] holds [I] to [src]..."), span_notice("You use [I] to locate where to cut off the chain reaction and attempt to stop it..."))
-		defuse()
+		defuse(forceperfect = FALSE)
```
Comment (ToasterBiome) [inline code/game/turfs/simulated/minerals.dm]:
```suggestion
		defuse(force_perfect = FALSE)
```

File: code/game/turfs/simulated/minerals.dm [hunk 6]
```diff
@@ -732,16 +736,19 @@
 			stage = GIBTONITE_DETONATE
 			explosion(bombturf,1,3,5, adminlog = notify_admins)
 
-/turf/closed/mineral/gibtonite/proc/defuse()
+/turf/closed/mineral/gibtonite/proc/defuse(forceperfect = FALSE)
```
Comment (ToasterBiome) [inline code/game/turfs/simulated/minerals.dm]:
```suggestion
/turf/closed/mineral/gibtonite/proc/defuse(force_perfect = FALSE)
```

File: code/game/turfs/simulated/minerals.dm [hunk 7]
```diff
@@ -732,16 +736,19 @@
...
 	if(stage == GIBTONITE_ACTIVE)
 		cut_overlay(activated_overlay)
 		activated_overlay.icon_state = "rock_Gibtonite_inactive"
 		add_overlay(activated_overlay)
 		desc = "An inactive gibtonite reserve. The ore can be extracted."
 		stage = GIBTONITE_STABLE
-		if(det_time < 0)
+		if(det_time < 0 && !forceperfect)
```
Comment (ToasterBiome) [inline code/game/turfs/simulated/minerals.dm]:
```suggestion
		if(det_time < 0 && !force_perfect)
```

File: code/game/turfs/simulated/minerals.dm [hunk 8]
```diff
@@ -732,16 +736,19 @@
...
+			det_time = 0
+			visible_message(span_notice("The chain reaction was stopped! The gibtonite had [det_time] reactions left till the explosion!"))
+		if(forceperfect)
```
Comment (ToasterBiome) [inline code/game/turfs/simulated/minerals.dm]:
```suggestion
		if(force_perfect)
```

File: code/game/turfs/simulated/minerals.dm [hunk 9]
```diff
@@ -732,16 +736,19 @@
...
 			det_time = 0
-		visible_message(span_notice("The chain reaction was stopped! The gibtonite had [det_time] reactions left till the explosion!"))
+			visible_message(span_notice("The chain reaction was stopped! The scanner ensured the gibtonite is at its highest potency!"))
```
Comment (ToasterBiome) [inline code/game/turfs/simulated/minerals.dm]: You are missing something here for if the det_time is > 0 and it's not forced perfect, using a normal mineral scanner here that you didn't get perfect will result in no message.

File: code/game/turfs/simulated/minerals.dm [hunk 10]
```diff
@@ -695,9 +696,12 @@
 	. = ..()
 
 /turf/closed/mineral/gibtonite/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/t_scanner/adv_mining_scanner/goat_scanner))
```
Comment (ToasterBiome) [inline code/game/turfs/simulated/minerals.dm]: needs a stage check like the next if statement, looks like you can defuse one that isn't going off...

File: code/game/turfs/simulated/minerals.dm [hunk 11]
```diff
@@ -695,9 +696,12 @@
...
+		user.visible_message(span_notice("[user] holds [I] to [src]..."), span_notice("[I] locates where to cut off the chain reaction and stops it."))
+		defuse(forceperfect = TRUE)
```
Comment (ToasterBiome) [inline code/game/turfs/simulated/minerals.dm]:
```suggestion
		defuse(force_perfect = TRUE)
```

**Review by ToasterBiome — CHANGES_REQUESTED** [review]

**Comment by LazennG** [comment]
remembered to add a feature @Chubbygummibear suggested where upon defusing a piece of gibtonite ore for the second time with the goat scanner, the ore's potency is raised if it isn't already high

**Review by nesisknesis — APPROVED** [review]

**Review by JamieD1 — APPROVED** [review]

**Comment by JamieD1** [comment]
Cornflakes

**Review by ToasterBiome — DISMISSED** [review]
remove shoes.dmi

**Review by JamieD1 — APPROVED** [review]

**Pull request merged**